numpy/core/src/multiarray/methods.c

```diff
@@ -582,14 +587,13 @@ array_tofile(PyArrayObject *self, PyObject *args, PyObject *kwds)
         return NULL;
     }
     if (PyBytes_Check(file) || PyUnicode_Check(file)) {
-        file = npy_PyFile_OpenFile(file, "wb");
```

Comment:
This looks wrong to me - `file` is a borrowed reference, but this now decrefs it.

Reply:
nevermind

Reply:
File is only borrowed until it is replaced by `NpyPath_PathlikeToFspath`